core/chain-configs/src/genesis_config.rs

```diff
@@ -468,37 +468,75 @@ impl GenesisJsonHasher {
     }
 }
 
+pub enum GenesisValidationMode {
+    Full,
+    UnsafeFast,
+}
+
 impl Genesis {
-    pub fn new(config: GenesisConfig, records: GenesisRecords) -> Self {
+    pub fn new(
+        config: GenesisConfig,
+        records: GenesisRecords,
+        genesis_validation: GenesisValidationMode,
```

Comment:
All calls to `new` pass in a literal `GenesisValidationMode` so perhaps this signature doesn't need to change?